app/uk/gov/hmrc/customs/notification/domain/pushNotificationRequest.scala

```diff
@@ -16,20 +16,41 @@
 
 package uk.gov.hmrc.customs.notification.domain
 
+import java.net.URL
 import java.util.UUID
 
 import play.api.libs.json.Format._
 import play.api.libs.json.{Format, Json, OFormat, _}
 import play.api.mvc.Headers
 
+import scala.util.Try
+
 case class Header(name: String, value: String)
 object Header {
   implicit val jsonFormat: OFormat[Header] = Json.format[Header]
 }
 
-case class PushNotificationRequestBody(url: String, authHeaderToken: String, conversationId: String,
+case class PushNotificationRequestBody(url: CallbackUrl, authHeaderToken: String, conversationId: String,
                                        outboundCallHeaders: Seq[Header], xmlPayload: String)
 object PushNotificationRequestBody {
+
+  implicit object CallbackUrlFormat extends Format[CallbackUrl] {
```

Comment:
Can the format code already in `DeclarantCallbackData` be used here rather than duplicating it?